src/gt4py/backend/debug_backend.py

```diff
@@ -56,47 +56,72 @@ def _make_regional_computation(self, iteration_order, interval_definition):
-        source_lines.extend(self._make_field_accessor(name, extent.to_boundary().lower_indices))
-
-        return source_lines
-
-    def make_stage_source(self, iteration_order: gt_ir.IterationOrder, regions: list):
+    def _make_ij_loop_lines(self, parallel_interval):
+        seq_axis_name = self.impl_node.domain.sequential_axis.name
+        axes_names = self.impl_node.domain.axes_names
         extent = self.block_info.extent
         lower_extent = extent.lower_indices
         upper_extent = extent.upper_indices
-        seq_axis_name = self.impl_node.domain.sequential_axis.name
-        axes_names = self.impl_node.domain.axes_names
 
         # Create IJ for-loops
         ij_loop_lines = []
-        for d in range(extent.ndims):
-            axis_name = axes_names[d]
-            if axis_name != seq_axis_name:
+        if not parallel_interval:
+            for d in range(extent.ndims):
+                axis_name = axes_names[d]
+                if axis_name != seq_axis_name:
+                    i = d + 1
+                    start_expr = "{:+d}".format(lower_extent[d]) if lower_extent[d] != 0 else ""
+                    size_expr = "{dom}[{d}]".format(dom=self.domain_arg_name, d=d)
+                    size_expr += " {:+d}".format(upper_extent[d]) if upper_extent[d] != 0 else ""
+                    range_expr = "range({args})".format(
+                        args=", ".join(a for a in (start_expr, size_expr, "") if a)
+                    )
+                    ij_loop_lines.append(
+                        " " * self.indent_size * i
+                        + "for {ax} in {range_expr}:".format(ax=axis_name, range_expr=range_expr)
+                    )
+        else:
+            for d in range(2):
                 i = d + 1
-                start_expr = "{:+d}".format(lower_extent[d]) if lower_extent[d] != 0 else ""
+                axis_name = axes_names[d]
+                assert len(parallel_interval) == 2
                 size_expr = "{dom}[{d}]".format(dom=self.domain_arg_name, d=d)
-                size_expr += " {:+d}".format(upper_extent[d]) if upper_extent[d] != 0 else ""
-                range_expr = "range({args})".format(
-                    args=", ".join(a for a in (start_expr, size_expr, "") if a)
-                )
+                axis_bound = parallel_interval[d]
+                bounds = []
+                for interval in (axis_bound.start, axis_bound.end):
+                    expr = f"{size_expr}" if interval.level == gt_ir.LevelMarker.END else "0"
+                    expr += "{:+d}".format(interval.offset) if interval.offset != 0 else ""
+                    bounds.append(expr)
+                range_expr = "range({args})".format(args=", ".join(bounds))
                 ij_loop_lines.append(
                     " " * self.indent_size * i
                     + "for {ax} in {range_expr}:".format(ax=axis_name, range_expr=range_expr)
                 )
 
```

Comment:
Is it possible to reduce code duplication between these two `if` clauses, possibly by using helper functions or parametrizing the differences between the blocks?